src/lib.rs

```diff
@@ -112,37 +149,22 @@ pub fn format_cargo_crates(packages: Vec<Package>, mode: AlignmentMode) -> Strin
     output
 }
 
-fn read_lockfiles(names: &Vec<String>) -> Result<Vec<Lockfile>> {
+/// Read the Cargo.lock files from the vector of filenames and resolve into
+/// a de-duplicated, sorted package vector.
+///
+/// This is a cargo2port internal function.
+pub fn read_packages_from_lockfiles(files: &Vec<String>) -> Result<Vec<Package>> {
```

Comment:
There are three options, because it's used in `src/main.rs` to transform filenames to packages. We could:

1. Leave it in the library crate (`src/lib.rs`) but remove the documentation or add `#[doc(hidden)]` to remove the documentation from what is published;
2. Move the function into the binary crate `src/main.rs`.
3. Remove the "internal" note.

Either will work, but while I wanted to write something that replaced this function (`crate2port`), it could be that someone else wants to write something that replaces the formatting output, so transforming a filename vector to a package vector might be something wanted as part of the fully public interface.

Reply:
I would prefer moving it the binary crate because `-` is specialized handling for CLI, and might not be suitable in a library.

Reply:
Done. This required changes to the test files, but I think that I'm happier with those changes anyway.

We aren't explicitly testing the new `lockfile_from_str` or `lockfile_from_stdio`, but the first is a wrapper around a `cargo_lock` function and the second seems like it would be *very hard* to test without some sort of forking and output capture and comparison.

I don't know how to do that in Rust.